core/services/bulletprooftxmanager/reaper.go

```diff
@@ -0,0 +1,140 @@
+package bulletprooftxmanager
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/store/orm"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"gorm.io/gorm"
+)
+
+//go:generate mockery --name ReaperConfig --output ./mocks/ --case=underscore
+
+// ReaperConfig is the config subset used by the reaper
+type ReaperConfig interface {
+	EthTxReaperThreshold() time.Duration
+	EthFinalityDepth() uint
+}
+
+// Reaper handles periodic database cleanup for BPTXM
+type Reaper struct {
+	db        *gorm.DB
+	config    ReaperConfig
+	log       *logger.Logger
+	blockNums utils.Mailbox
+
+	chStop chan struct{}
+	chDone chan struct{}
+}
+
+// NewReaper instantiates a new reaper object
+func NewReaper(db *gorm.DB, config ReaperConfig) *Reaper {
+	return &Reaper{
+		db,
+		config,
+		logger.CreateLogger(logger.Default.With("id", "bptxm_reaper")),
+		*utils.NewMailbox(1),
+		make(chan struct{}),
+		make(chan struct{}),
+	}
+}
+
+// Start the reaper. Should only be called once.
+func (r *Reaper) Start() {
+	r.log.Debugf("EthTxReaper: started with age threshold %v", r.config.EthTxReaperThreshold())
+	go r.runLoop()
+}
+
+// Stop the reaper. Should only be called once.
+func (r *Reaper) Stop() {
+	r.log.Debug("EthTxReaper: stopping")
+	close(r.chStop)
+	<-r.chDone
+}
+
+func (r *Reaper) runLoop() {
+	defer close(r.chDone)
+	for {
+		select {
+		case <-r.chStop:
+			return
+		case <-r.blockNums.Notify():
+			n, exists := r.blockNums.Retrieve()
+			if !exists {
+				continue
+			}
+			latestBlockNum := n.(int64)
+			err := r.ReapEthTxes(latestBlockNum)
+			if err != nil {
+				r.log.Error("Reaper: unable to reap old eth_txes: ", err)
+			}
+
+		}
+	}
+}
+
+// NewHighestBlockNum should be called on every new highest block number
+func (r *Reaper) NewHighestBlockNum(latestBlockNum int64) {
```

Comment:
oh is this called by anything right now? Can't see any usages.

Reply:
Good catch, forgot to actually add the call in EthConfirmer. Lemme push in a sec ;)